File: quarkus-jdbi/src/main/java/com/github/frederikpietzko/dto/PetDto.java
package com.github.frederikpietzko.dto;

import com.github.frederikpietzko.model.Pet;
import com.github.frederikpietzko.model.Type;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

public record PetDto(
  @NotNull Long id,
  @NotEmpty String name,
  @Min(1) int age,
  @NotNull Type type
) {
  public static PetDto fromEntity(Pet entity) {
    return new PetDto(entity.getId(), entity.getName(), entity.getAge(), entity.getType());
  }
}

File: http4k/src/main/kotlin/com/github/frederikpietzko/dto/PetDto.kt
package com.github.frederikpietzko.dto

import kotlinx.serialization.Serializable

@Serializable
data class PetDto(
  val id: Long,
  val name: String,
  val age: Int,
  val type: String,
)

fun com.github.frederikpietzko.model.Pet.toDto() = PetDto(
  id = id!!,
  name = name,
  age = age,
  type = type.name,
)
